java/ql/lib/semmle/code/java/frameworks/Strings.qll

```diff
@@ -11,19 +11,28 @@ private class StringSummaryCsv extends SummaryModelCsv {
         "java.lang;String;false;concat;(String);;Argument[0];ReturnValue;taint",
         "java.lang;String;false;concat;(String);;Argument[-1];ReturnValue;taint",
         "java.lang;String;false;copyValueOf;;;Argument[0];ReturnValue;taint",
-        "java.lang;String;false;endsWith;;;Argument[-1];ReturnValue;taint",
+        "java.lang;String;false;describeConstable;;;Argument[-1];Element of ReturnValue;value",
         "java.lang;String;false;format;(Locale,String,Object[]);;Argument[1];ReturnValue;taint",
         "java.lang;String;false;format;(Locale,String,Object[]);;ArrayElement of Argument[2];ReturnValue;taint",
         "java.lang;String;false;format;(String,Object[]);;Argument[0];ReturnValue;taint",
         "java.lang;String;false;format;(String,Object[]);;ArrayElement of Argument[1];ReturnValue;taint",
         "java.lang;String;false;formatted;(Object[]);;Argument[-1];ReturnValue;taint",
         "java.lang;String;false;formatted;(Object[]);;ArrayElement of Argument[0];ReturnValue;taint",
-        "java.lang;String;false;getChars;;;Argument[-1];Argument[2];taint",
         "java.lang;String;false;getBytes;;;Argument[-1];ReturnValue;taint",
+        "java.lang;String;false;getBytes;(int,int,byte[],int);;Argument[-1];Argument[2];taint",
+        "java.lang;String;false;getChars;;;Argument[-1];Argument[2];taint",
         "java.lang;String;false;indent;;;Argument[-1];ReturnValue;taint",
         "java.lang;String;false;intern;;;Argument[-1];ReturnValue;taint",
-        "java.lang;String;false;join;;;Argument[0..1];ReturnValue;taint",
+        "java.lang;String;false;join;;;Argument[0..1];ReturnValue;taint", // TODO: ArrayElement of Argument?
+        "java.lang;String;false;lines;;;Argument[-1];Element of ReturnValue;taint",
         "java.lang;String;false;repeat;(int);;Argument[-1];ReturnValue;taint",
+        "java.lang;String;false;replace;;;Argument[1];ReturnValue;taint",
+        "java.lang;String;false;replace;;;Argument[-1];ReturnValue;taint",
+        "java.lang;String;false;replaceAll;;;Argument[1];ReturnValue;taint",
+        "java.lang;String;false;replaceAll;;;Argument[-1];ReturnValue;taint",
+        "java.lang;String;false;replaceFirst;;;Argument[1];ReturnValue;taint",
+        "java.lang;String;false;replaceFirst;;;Argument[-1];ReturnValue;taint",
```

Comment:
These ones have non-CSV models that should be cleaned up concurrently (search for `"replaceAll"` for example)